### PR TITLE
feat: [CDS-52543]: file-store-open-from-environment-items

### PR DESCRIPTION
**Pull request by litvinovstas**
### Summary
Added option to open file store from environments items (Manifests/Config files/Application config)

Screenshot 2023-02-06 at 15 46 47
Screenshot 2023-02-06 at 15 47 39

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by litvinovstas** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/LocationValue.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        if (isTooltip) {
+          return (
+            <Text
+              color={Color.BLACK}
+              padding="small"
+              className={cx(isHarnessFileStore && css.locationLink)}
+              key={`${locationValue}${i}`}
+              onClick={() => handleClick(locationValue)}
+            >
+              {`${locationValue}`}
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/LocationValue.tsx:44]: We don't need the string literal here

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/LocationValue.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  return (
+    <>
+      {locations.map((locationValue: string, i: number) => {
+        if (isTooltip) {
+          return (
+            <Text
+              color={Color.BLACK}
+              padding="small"
+              className={cx(isHarnessFileStore && css.locationLink)}
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/LocationValue.tsx:40]: Let's not import cx just for this. We can use ternary operator instead

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesSelection.module.scss [hunk 3]
```diff
@@ -103,3 +103,6 @@
 .locationLink {
   cursor: pointer;
 }
+.tiptoolLink {
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesSelection.module.scss:106]: tooltip?

**Comment by litvinovstas** [comment]
retrigger jest

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/LocationValue.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  return (
+    <>
+      {locations.map((locationValue: string, i: number) => {
+        if (isTooltip) {
+          return (
+            <Text
+              color={Color.BLACK}
+              padding="small"
+              className={isHarnessFileStore ? css.locationLink : ''}
+              key={`${locationValue}${i}`}
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/LocationValue.tsx:40]: It could have been {locationValue + i}

**Review by AshwinBhatkal-Harness — APPROVED** [review]

File: src/modules/70-pipeline/components/ManifestSelection/ManifestListView/ManifestListView.tsx [hunk 5]
```diff
@@ -588,28 +590,30 @@ function ManifestListView({
...
                           >
-                            {typeof manifestLocation === 'string' ? manifestLocation : manifestLocation.join(', ')}
+                            {typeof manifestLocation === 'string' ? (
+                              manifestLocation
+                            ) : (
+                              <LocationValue
+                                locations={manifestLocation}
+                                isHarnessStore={manifest?.spec?.store.type === ManifestStoreMap.Harness}
+                                onClick={(path: string, scope: string) =>
+                                  FileStoreModal.openFileStoreModal(path, scope)
```
Comment (vikrant-harness) [inline src/modules/70-pipeline/components/ManifestSelection/ManifestListView/ManifestListView.tsx:613]: Can we extract this out as we are passing the same params in both of them

**Review by vikrant-harness — APPROVED** [review]

**Comment by litvinovstas** [comment]
retrigger jest

**Comment by litvinovstas** [comment]
retrigger jest

**Comment by litvinovstas** [comment]
retrigger cypress-cv

**Comment by litvinovstas** [comment]
retrigger cypress-pipeline

**Comment by litvinovstas** [comment]
retrigger cypress-pipeline

**Review by vikrant-harness — APPROVED** [review]

**Review by vkbansal-harness — APPROVED** [review]

**Pull request merged**